### PR TITLE
fix(ooi_report): take description from finding_type if None in Finding

### PR DESCRIPTION
**Pull request by Lisser**
## Changes
Fixes:
- Crash when `finding.description == None` in Octopoes. Replace description with description from OOIInformation
- error handling when PDF cannot be generated. Show an error message instead of returning a corrupt PDF file

## Issue ticket number and link
-

## Proof
Screenshot 2023-01-05 at 15 10 07

## Good to know
- Does this PR introduce or depend on API-incompatible changes? If yes: what do other users/developers need to do or confirm before merging?
  - No
- Does this PR depend on a specific version of a library?
  - No
- Does this PR require config, setup, or `.env` changes?
  - No

## Checklist for author(s):
- [x] This PR comes from a `feature` or `hotfix` branch, in line with our git branching strategy;
- [x] I have squashed all commits before merging this PR;
- [x] This PR is "bite-sized" and only focuses on a single issue, problem, or feature;
- [x] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [x] If a non-trivial PR: I have added screenshots or some other proof that my code does what it is supposed to do;
- [x] I am not reinventing the wheel: there is no high-quality library that already has this feature;
- [x] I have changed the example `.env` files if I added, removed, or changed any config options, and I have informed others that they need to modify their `.env` files if required;
- [x] I have performed a self-review of my own code;
- [x] I have commented my code, particularly in hard-to-understand areas;
- [x] I have made corresponding changes to the documentation, if necessary;
- [x] I have written unit, integration, and end-to-end tests for the change that I made;

```
## Checklist for functional reviewer(s):
- [ ] If a non-trivial PR: This PR is properly linked to an issue on the project board;
- [ ] I have checked out this branch, and successfully ran `make kat`;
- [ ] I have ran `make test-rf` and all end-to-end Robot Framework tests pass;
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended;
- [ ] I confirmed that there are no unintended functional regressions in this branch;

### What works:
* _bullet point + screenshot (if useful) per tested functionality_

### What doesn't work:
* _bullet point + screenshot (if useful) per tested functionality_

### Bug or feature?:
* _bullet point + screenshot (if useful) if it is unclear whether something is a bug or an intended feature._
```

```
## Checklist for code reviewer(s):
- [ ] The code passes the CI tests and linters;
- [ ] The code does not bypass authentication or security mechanisms;
- [ ] The code does not introduce any dependency on a library that has not been properly vetted;
- [ ] The code does not violate Model-View-Template and our other architectural principles;
- [ ] The code contains docstrings, comments, and documentation where needed;
- [ ] The code prioritizes readability over performance where appropriate;
- [ ] The code conforms to our agreed coding standards.
```

### CODE REVIEW

File: rocky/keiko.py [hunk 1]
```diff
@@ -24,17 +24,16 @@ def generate_report(self, template: str, data: Dict, glossary: str) -> str:
...
         for i in range(15):
+            time.sleep(1)
             res = self.session.get(f"{self._base_uri}/reports/{report_id}.keiko.pdf")
             if res.status_code == 200:
                 return res.content
-            else:
-                time.sleep(1)
-
-        return None
+            if i == 14:
```
Comment (ammar92) [inline rocky/keiko.py:35]: This `if` is redundant, you can make the `res.raise_for_status()` call at the end of the body after the loop finishes

File: rocky/views/ooi_report.py [hunk 2]
```diff
@@ -227,12 +226,16 @@ def get(self, request, *args, **kwargs):
         report_file_name = f"{report_file_name}.pdf"
 
         # open pdf as attachment
-        response = HttpResponse(
-            keiko_client.get_report(report_id),
-            content_type="application/pdf",
-        )
-        response["Content-Disposition"] = f'attachment; filename="{report_file_name}"'
-        return response
+        try:
```
Comment (ammar92) [inline rocky/views/ooi_report.py:229]: See https://docs.djangoproject.com/en/3.2/ref/request-response/#fileresponse-objects

**Review by ammar92 — COMMENTED** [review]
Nice tests. Please see also remarks

File: rocky/keiko.py (hunk 1 above)
Comment (Lisser) [reply rocky/keiko.py:35]: Yeah, it's ugly, sorry. Ran into this warning and wasn't sure about the cleanest solution...

Screenshot 2023-01-06 at 10 00 52

File: tests/test_ooi_report.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    assert response.status_code == 200
+    assert response.content == b"fake_binary_pdf_content"
+
+    report_data_param = mock_keiko_client.generate_report.call_args[0][1]
+    # Verify that the data is sent correctly to Keiko
+    assert report_data_param["meta"]["total"] == 1
+    assert report_data_param["meta"]["total_by_severity"]["low"] == 1
+    assert report_data_param["meta"]["total_by_severity_per_finding_type"]["low"] == 1
+    assert report_data_param["meta"]["total_by_finding_type"]["KAT-000"] == 1
+    assert report_data_param["meta"]["total_finding_types"] == 1
```
[the remark this answers is not in the document]
Comment (Lisser) [reply tests/test_ooi_report.py:128]: Yes. Done, thanks

File: rocky/keiko.py (hunk 1 above)
Comment (ammar92) [reply rocky/keiko.py:35]: In that case you could use the `else` clause:
```python3
for ...:
    pass
else:
    pass  # executed when the loop is completed without any break
```

Comment (Lisser) [reply rocky/keiko.py]: Fixed.

File: rocky/views/ooi_report.py (hunk 2 above)
Comment (Lisser) [reply rocky/views/ooi_report.py:229]: Fixed

**Review by ammar92 — APPROVED** [review]

**Pull request merged**